### PR TITLE
111 lesson amount update after adding

### PR DESCRIPTION
**Pull request by potapchuk-anna**
https://trello.com/c/OAvjYw4V/111-created-lesson-and-total-quantity-of-lessons-increased-on-1-appear-on-timetable-only-after-refreshing-page

### CODE REVIEW

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html [hunk 1]
```diff
@@ -7,12 +7,7 @@ <h2 class="date">{{ todayDate }}</h2>
 
     <div class="lessons-section" *ngIf="lessons">
         <div class="lessons-column">
-            <div class="lesson" *ngFor="let lesson of lessonsColumn1">
```
Comment (ShevchenkoKirilo) [inline frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html:10]: this thing is mandatory
image

**Review by ShevchenkoKirilo — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html (hunk 1 above)
Comment (ShevchenkoKirilo) [reply frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html:10]: Return 2 columns, it's the only way to display cards with different height without white space

Comment (potapchuk-anna) [reply frontend/src/app/modules/lessons/lessons-page/lessons-page.component.html:10]: ok

**Review by ShevchenkoKirilo — DISMISSED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
 
     public async Task<ICollection<DayCardDto>?> GetDayCardsOfWeekAsync(RequestDayCardDto requestDto)
     {
-        var mondayDate = requestDto.Date.AddDays(-(int)requestDto.Date.DayOfWeek).Date;
+        var mondayDate = requestDto.Date.AddDays(-((7 + (requestDto.Date.DayOfWeek - DayOfWeek.Monday)) % 7)).Date;
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:61]: Why you need 7 here?
7%7=0

File: frontend/src/app/modules/filter-section/filter-section/filter-section.component.html [hunk 3]
```diff
@@ -5,7 +5,7 @@
     </div>
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/modules/filter-section/filter-section/filter-section.component.html:5]: You have a bit too much changes in PR, do you want to fix all of them?

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs (hunk 2 above)
Comment (potapchuk-anna) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:61]: Sunday = 0 
Monday = 1
[src](https://learn.microsoft.com/en-us/dotnet/api/system.dayofweek?view=net-7.0)
If current daty is Sunday: (0 - 1) % 7 = -1
In my implementation: (7 + 0 - 1) % 7 = 6

Comment (tatianahutii) [reply backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:73]: This statement looks strange, please simplify it

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs [hunk 4]
```diff
@@ -87,6 +87,15 @@ public async Task<ICollection<LessonDto>> GetAllLessonsAsync(FiltersRequest filt
...
     }
 
+    private DateTime GetMondayDate(DateTime date)
+    {
+        if(date.DayOfWeek == DayOfWeek.Sunday)
+        {
+            return date.AddDays(6).Date;
+        }
+        return date.AddDays(-((date.DayOfWeek - DayOfWeek.Monday) % 7)).Date;
+    }
```
Comment (Anatolii-Ivanov) [inline backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/LessonsService.cs:97]: I guess something is wrong)
1. Sunday is 6
2. All other days are -5...0

And do you still need %7?
+ I will move minus to GetDayCardsOfWeekAsync function

**Review by Anatolii-Ivanov — DISMISSED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

File: frontend/src/app/modules/lessons/lessons-page/lessons-page.component.sass [hunk 5]
```diff
@@ -52,7 +52,7 @@
 .lessons-column
     display: flex
     flex-direction: column
-    width: calc( (100% - 55px) / 2 )
```
[the remark this answers is not in the document]
Comment (potapchuk-anna) [reply frontend/src/app/modules/lessons/lessons-page/lessons-page.component.sass:55]: Sorry, there were two width in this code and I selected wrong...

**Pull request merged**